### PR TITLE
add a script that should help find mappings and add a few mappings

### PR DESCRIPTION
**Pull request by dgcampbe**
This should help solve the issue with #30

### CODE REVIEW

File: icon_search.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# This script searches for icons on your system by name to help find entries for mapping.yaml
+
+echo -n "Please enter an app to search for: "
+read app # Store the prompted app name
+declare -a paths
+readarray -t paths < <(find /usr/share/icons/ -name *$app*.*) # Make an array storing every result for the given app name
```
Comment (autinerd) [inline icon_search.sh:8]: Maybe we should search in all possible locations (especially the icons folder in the home directory)

File: icon_search.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (sertonix) [inline icon_search.sh:1]: To be consistent with the other scripts this should be 
```suggestion
#!/usr/bin/env bash
```

**Comment by dgcampbe** [comment]
I overhauled the script to have much more helpful output. :smile:

File: icon_search.sh (hunk 1 above)
Comment (dgcampbe) [reply icon_search.sh]: Do you mean `~/.icons`?

Comment (autinerd) [reply icon_search.sh]: `~/.local/share/icons` is the offical location for user-installed icons.

**Comment by dgcampbe** [comment]
added search to ~/.local/share/icons and a few other improvements

**Comment by autinerd** [comment]
If you want, we could also use another language like Python :smile:

Some things that should be done:
- sort entries
- remove duplicates
- some themes like Breeze have the folder structure `<icon category>/<size>/<icon name>` (like `apps/24/kwrite.svg`)

**Comment by dgcampbe** [comment]
I choose bash for consistency since there are already some bash scripts in the repo, but rewriting it in python would definitely make it more legible and less annoying to write. I would be happy to do the rewrite in python, if you would like. I at least learned a lot of bash in the process.

File: mapping.yaml [hunk 3]
```diff
@@ -26,6 +26,12 @@ appearance:
 ardour:
   - apps/ardour8
 arduinodroid:
+  - apps/arduino
+  - apps/arduino-icon-small
+  - apps/arduino-arduinoide
+  - apps/arduino-ide
+  - apps/arduino-ide-v2-symbolic
```
Comment (autinerd) [inline mapping.yaml:33]:
```suggestion
  - apps/arduino-ide-v2
```

File: mapping.yaml [hunk 4]
```diff
@@ -522,6 +529,7 @@ letter_lowercase_square_g:
   - apps/steam_icon_4000
 letter_uppercase_circle_p:
   - apps/polyglot
+  - apps/polyglot-symbolic
```
Comment (autinerd) [inline mapping.yaml:532]:
```suggestion
```

File: mapping.yaml [hunk 5]
```diff
@@ -967,6 +978,10 @@ terminal:
   - apps/xterm-color_48x48
   - apps/yakuake
   - system/konsole
+  - apps/xfce-terminal-symbolic
```
Comment (autinerd) [inline mapping.yaml:981]:
```suggestion
  - apps/xfce-terminal
```

File: mapping.yaml [hunk 6]
```diff
@@ -1053,3 +1068,10 @@ yuzu:
   - apps/yuzu
 zoom:
   - apps/zoom-desktop
+beeper:
+  - apps/beeper
+blender:
+  - apps/blender
+  - apps/blender-symbolic
+  - apps/blender-2.7-symbolic
```
Comment (autinerd) [inline mapping.yaml:1076]:
```suggestion
```

**Review by autinerd — CHANGES_REQUESTED** [review]
The "-symbolic" suffixes should be removed.

**Comment by autinerd** [comment]
I have now reworked the Python script :sweat_smile: It now parses the theme files, which is much faster than the way of doing a recurse glob.

**Comment by autinerd** [comment]
If you want, we can delete the shell file and only use the Python script.

A future step would be to evaluate the .desktop files as well.

**Comment by dgcampbe** [comment]
Ha! I didn't know you there were theme files. That is a massively better way of doing. Thanks!

**Review by autinerd — APPROVED** [review]

**Comment by autinerd** [comment]
Thank you very much!

**Pull request merged**